connectors/es/sink.py

```diff
@@ -78,6 +78,13 @@ class ContentIndexDoesNotExistError(Exception):
     pass
 
 
+class ElasticsearchOverloadedError(Exception):
+    def __init__(self, cause=None):
+        msg = "Connector was unable to ingest data into overloaded Elasticsearch. Make sure Elasticsearch instance is healthy, has enough resources and content index is healthy."
```

Comment:
Can we include the index name here? "[...] has enough resources and content index ({index_name}) is healthy"

Reply:
There's actually no way to do this with current design - Sink does not operate on an index, rather it operates on "operations" abstraction. I can try to get data from the first operation that's left there and get index name, but rather not.

I'm planning to refactor this code more in next release to allow better transparent retries, and this error message might get improved on the way.

Reply:
I think we pass the target index [here](https://github.com/elastic/connectors/blob/main/connectors/es/sink.py#L770) (`async_bulk`), right? So we could simply pass that to `Sink`, too or do we've cases where the target indices differ for bulk operations of the same `Sink` instance?

But ofc nothing blocking, if you think we need some refactoring first, I'm also happy with that.

Reply:
Sink can operate on multiple indices, so passing a single index into it feels like something that breaks design.

We don't have cases when target indices are different, so I can infer which index is being used, but same can be done in UI for now by simply seeing an error message and name of index.

I'd love to put specific name into the message, but current class design is not allowing for that, I want to address it in later work for next release.